File: samples/CSharp/Northwind/Shipper.cs
using System;
using System.Collections.ObjectModel;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Shippers")]
   public class Shipper {

      [Column(IsPrimaryKey = true, IsDbGenerated = true)]
      public int ShipperID { get; set; }

      [Column]
      public string CompanyName { get; set; }

      [Column]
      public string Phone { get; set; }

      [Association(OtherKey = nameof(Order.ShipVia))]
      public Collection<Order> Orders { get; } = new Collection<Order>();
   }
}

File: samples/VisualBasic/Northwind/Shipper.vb
Imports System
Imports System.Collections.ObjectModel
Imports DbExtensions

Namespace Northwind

   <Table(Name:="Shippers")>
   Public Class Shipper

      <Column>
      Property CompanyName As String

      <Column>
      Property Phone As String

      <Column(IsPrimaryKey:=True, IsDbGenerated:=True)>
      Property ShipperID As Integer

      <Association(OtherKey:=NameOf(Order.ShipVia))>
      ReadOnly Property Orders As New Collection(Of Order)

   End Class

End Namespace
